src/github.com/getlantern/peerscanner/web.go

```diff
@@ -51,36 +56,46 @@ func startHttp() {
 // register is the entry point for peers registering themselves with the service.
 // If peers are successfully vetted, they'll be added to the DNS round robin.
 func register(resp http.ResponseWriter, req *http.Request) {
-	name, ip, port, err := getHostInfo(req)
-	if err == nil && port != 443 {
-		err = fmt.Errorf("Port %d not supported, only port 443 is supported", port)
+	name, ip, port, supportedFronts, err := getHostInfo(req)
+	if err == nil && !(port == "80" || port == "443") {
+		err = fmt.Errorf("Port %d not supported, only ports 80 and 443 are supported", port)
 	}
 	if err != nil {
 		resp.WriteHeader(http.StatusBadRequest)
 		fmt.Fprintln(resp, err.Error())
 		return
 	}
-
-	online := true
-	connectionRefused := false
-	timedOut := false
-
 	if isPeer(name) {
 		log.Debugf("Not adding peer %v because we're not using peers at the moment", name)
-	} else {
-		h := getOrCreateHost(name, ip)
-		online, connectionRefused, timedOut = h.status()
-		if timedOut {
-			log.Debugf("%v timed out waiting for status, returning 500 error", h)
-			resp.WriteHeader(500)
-			fmt.Fprintf(resp, "Timed out waiting for status")
-			return
-		}
+		resp.WriteHeader(200)
+		fmt.Fprintln(resp, "Peers disabled at the moment")
+		return
 	}
+	online := true
+	connectionRefused := false
+	timedOut := false
 
+	h := getOrCreateHost(name, ip, port)
+	online, connectionRefused, timedOut = h.status()
 	if online {
 		resp.WriteHeader(200)
 		fmt.Fprintln(resp, "Connectivity to proxy confirmed")
+		if (supportedFronts & cloudfrontBit) == cloudfrontBit {
+			h.initCloudfront()
```

Comment:
Right -- I was just asking if that's where the 15 minute Cloudfront distribution kickoff takes place, not that it would ever block for 15 minutes (good god!).